### DISCUSSION TITLE
How do I change the ~/.m2 directory?

### DISCUSSION THREAD

**Discussion by styfle** [thheller/shadow-cljs#880] [Q&A] [answered]
I noticed that running the hello world `npx create-cljs-project` and then `npx shadow-cljs compile app` it creates a `.shadow-cljs` directory in my current project directory, but it creates a `.m2` directory in my HOME directory.

Is there a way to change this behavior so these files are created elsewhere? Perhaps with an environment variable or CLI flag?

**Comment by thheller** [comment, marked as answer]
You can configure `:maven {:local-repo "foo"}` in `shadow-cljs.edn` to use the `foo` directory in your project folder. It is however common to keep the `.m2` directory since multiple projects may share these files.

**Reply by styfle** [reply]
Thanks for the quick response!

~~Does "foo" have to be a static string or can I somehow make it "use the current working directory"? Basically the same place that `.shadow-cljs` lives?~~

Oh I see, it is relative to the cwd already 🤦  It worked!

**Reply by styfle** [reply]
> common to keep the .m2 directory since multiple projects may share these files.

Yep I agree for local development. I would like to keep this `shadow-cljs.edn` file as-is for development but configure with an environment variable for CI so the m2 directory can be cached between runs.

Is that possible?

**Reply by thheller** [reply]
You can use `:maven {:local-repo #shadow/env ["FOO" "foo"]}`. That'll use the `FOO` environment variable when set, otherwise just `foo`

**Reply by styfle** [reply]
Not bad, thanks!

The disadvantage here is that every shadow-cljs has to add this configuration so I was hoping for a solution that didn't require modifying every config.

Should I create a feature request? 😄

**Reply by thheller** [reply]
I'm not sure what your intention is here. You may also configure this in ~/.shadow-cljs/config.edn but I recommend sticking with the default.

**Reply by styfle** [reply]
I'm looking at improving CI build times.

The intention is for a single CI config that will build and cache any shadow-cljs project without changing existing project files.

So thats why I was wondering if I could configure with an environment variable, so I wound't need to ask other users to modify their source files.

I'm looking for behavior similar to Node's [npm](https://www.npmjs.com) (node_modules dir) or Ruby's [bundler](https://bundler.io) (vendor dir).

**Reply by thheller** [reply]
The common method is to have your CI system cache the m2 folder. Unsure which users you are referring to here? I assumed you are talking about your setup?
